src/map/Pins.js

```diff
@@ -29,47 +33,143 @@
     Pins.prototype.layerUpdate = function (base) {
         Layer.prototype.layerUpdate.apply(this, arguments);
 
-        this._pinsTransform
-            .style("opacity", this.opacity())
-        ;
-
         this.pinsPaths = this._pinsTransform.selectAll(".data").data(this.visible() ? this.data() : [], function (d) { return d[0]; });
         var context = this;
-        this.pinsPaths.enter().append("path")
-            .attr("class", "data")
-            .attr("d", function (d) {
-                return "M 0,0 C -2,-20 -10,-22 -10,-30 A 10,10 0 1,1 10,-30 C 10,-22 2,-20 0,0 z M -2,-30";
-            })
-            .call(this._selection.enter.bind(this._selection))
-            .on("click", function (d) {
-                context.click(context.rowToObj(d), "weight", context._selection.selected(this));
-            })
-        ;
-        this.pinsPaths
-            .attr("stroke-width", 1.5 + "px")
-            .style("display", function (d) {
-                var pos = base.project(d[0], d[1]);
-                if (!pos) {
-                    return "none";
+        this.pinsPaths.enter().append("g").each(function(d) {
+
+            d3.select(this).style("opacity", d[2] && d[2].opacity ? d[2].opacity : context.opacity()); // might need to rework this from where it was using pinstransform?
+
+            if (!d[2] || d[2].pinShape !== "none") {
+                var path = d3.select(this).append("path")
+                    .attr("class", "data")
```

Comment:
@GordonSmith is this really such a good class name for this .... maybe pin-path or pin-shape ?